pkg/policy/l4.go

```diff
@@ -144,7 +144,7 @@ func (l4 *L4Filter) ToKeys(direction trafficdirection.TrafficDirection, identity
 	port := uint16(l4.Port)
 	proto := uint8(l4.U8Proto)
 
-	if l4.AllowsAllAtL3() {
+	if l4.AllowsAllAtL3() && l4.Port != 0 {
```

Comment:
If L4.Port is 0 and it allows all at L3, won't this generate BPF entries with (0, 0)? The datapath won't respect entries like that right now. Either we need to change the datapath to support this or we should avoid making this change, so that we still go through all the identities and generate the L3 allows for all peer identities.

Reply:
With a bit more thinking here, it's not that bad if we generate this entry so long as it doesn't shortcut the generation of entries with non-zero identity + zero port (which I suspect it may end up doing in this case). If we do decide to leave this in place, it might help to have a comment describing that as of the changes at this point, `__policy_can_access()` in BPF does not respect it.

Reply:
No, the logic skips this wildcard generate if the L4 Port is 0; we explicitly do not want to generate BPF entries with identity = 0 and port = 0 because this is not respected by the datapath. This is because we don't handle wildcarding at L3 in the datapath with a value of zero for the port. Did you read this as `if allows all at L3 and port = 0` instead of `if allows all at l3 and port != 0`? Your  statement leads me to think that you did. I can add a comment indicating why we aren't generating this key here.

Reply:
Oh, weird, you've astutely noticed that I just completely mixed up the previous code vs. the new code, I read the diff as the opposite. Looking again this looks fine. Apologies for the noise!